shared/src/main/java/uk/gov/di/authentication/shared/services/DynamoService.java

```diff
@@ -320,6 +320,21 @@ public void updateMFAMethod(
                         .setMfaMethod(mfaMethod));
     }
 
+    @Override
+    public void setMFAMethodVerifiedTrue(String email, MFAMethodType mfaMethodType) {
+        String dateTime = NowHelper.toTimestampString(NowHelper.now());
+        var userCredentials =
+                userCredentialsMapper.load(UserCredentials.class, email.toLowerCase(Locale.ROOT));
+        var mfaMethod =
+                userCredentials.getMfaMethods().stream()
+                        .filter(
+                                method ->
+                                        method.getMfaMethodType().equals(mfaMethodType.getValue()))
+                        .findFirst();
+        mfaMethod.ifPresent(method -> method.setMethodVerified(true));
```

Comment:
This could just fail quitely if in the unlikely event no mfaMethod is found and therefore it will not set verified to true. But the lambda will still return a 204. 

I would assume it is present as you're already checking it is in the AuthCodeAppvalidator and throw if it isn't.